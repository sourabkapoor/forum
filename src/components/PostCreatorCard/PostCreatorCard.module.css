.post-creator-card {
  width: 100%;
  color: black;
  display: flex;
  align-items: center;
}

.post-creator-card__input {
  background-color: white;
  border-radius: 4px;
  border: 1px solid #343536;
  box-sizing: border-box;
  height: 38px;
  margin-right: 8px;
  outline: none;
  padding: 0 16px;
  color: var(--secondary-color);
  flex-grow: 1;
}

.post-creator-card__icon-link {
  border-radius: 4px;
  min-height: 40px;
  min-width: 40px;
  display: grid;
  place-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
}

/* .post-creator-card__icon-link:hover {
  background-color: #343536;
} */

.post-creator-card__icon {
  color: #818384;
  width: 24px;
  height: 24px;
  display: inline-block;
  line-height: 20px;
  font-size: 20px;
  border-radius: 4px;
}
.post-creator-card__picture {
  /* background-color: #343536; */
  border-radius: 50%;
  margin-right: 8px;
  flex-basis: 38px;
  cursor: pointer;
}
.post-creator-card__profile-icon {
  width: 38px;
  height: 38px;
  transform: scale(1.3);
  transform-origin: bottom center;
  clip-path: polygon(
    0 68.22%,
    12.12% 68.22%,
    12.85% 71.49%,
    13.86% 74.69%,
    15.14% 77.79%,
    16.69% 80.77%,
    18.49% 83.6%,
    20.54% 86.26%,
    22.8% 88.73%,
    25.28% 91%,
    27.94% 93.04%,
    30.77% 94.85%,
    33.75% 96.4%,
    36.85% 97.68%,
    40.05% 98.69%,
    43.32% 99.42%,
    46.65% 99.85%,
    50% 100%,
    53.35% 99.85%,
    56.68% 99.42%,
    59.95% 98.69%,
    63.15% 97.68%,
    66.25% 96.4%,
    69.23% 94.85%,
    72.06% 93.04%,
    74.72% 91%,
    77.2% 88.73%,
    79.46% 86.26%,
    81.51% 83.6%,
    83.31% 80.77%,
    84.86% 77.79%,
    86.14% 74.69%,
    87.15% 71.49%,
    87.88% 68.22%,
    100% 68.22%,
    100% 0,
    0 0
  );
}

@media (max-width: 640px) {
  .post-creator-card {
    width: 100%;
  }
}
