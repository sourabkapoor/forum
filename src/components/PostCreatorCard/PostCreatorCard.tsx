/* eslint-disable jsx-a11y/img-redundant-alt */
import CSSModules from "react-css-modules";
import Card from "../Card/Card";
import styles from "./PostCreatorCard.module.css";
import { AiOutlinePicture } from "react-icons/ai";
import { BsLink45Deg } from "react-icons/bs";
import { useRef } from "react";
import { useNavigate } from "react-router-dom";
import { useAppSelector } from "../../hooks/hooks";
import { selectCommunityData } from "../../features/subreddit/subredditSlice";
import { selectAuthStatus } from "../../features/auth/authSlice";
import AccountCircleOutlinedIcon from '@mui/icons-material/AccountCircleOutlined';

const PostCreatorCard: React.FC = () => {
  const inputRef = useRef<HTMLInputElement | null>(null);

  const isLoggedIn = useAppSelector(selectAuthStatus);
  const { name } = useAppSelector(selectCommunityData);

  const navigate = useNavigate();

  function inputFocus() {
    navigate(`/r/${name}/submit`);
  }

  function navigateToUser() {
    navigate("/user");
  }

  function navigateToCreatePostPage() {
    navigate(`/r/${name}/submit`);
  }
  return (
    <>
      {isLoggedIn && (
        <Card>
          <div styleName="post-creator-card">
            <div
              styleName="post-creator-card__picture"
              onClick={navigateToUser}
              data-testid="profile-picture"
            >
              <AccountCircleOutlinedIcon style={{fontSize: '28px'}} />
              {/* <img
                styleName="post-creator-card__profile-icon"
                src="https://styles.redditmedia.com/t5_g2km0/styles/profileIcon_snood9e1c00c-b1c6-46a9-a231-a6fcd78cdd16-headshot.png?width=256&height=256&frame=1&crop=256:256,smart&s=7bd48be150588ab7dc3dc5c73be4d0dcddeeae8d"
                alt="default reddit profile"
              /> */}
            </div>
            <input
              ref={inputRef}
              styleName="post-creator-card__input"
              placeholder="Create Post"
              onFocus={inputFocus}
            />
            <button
              styleName="post-creator-card__icon-link"
              onClick={navigateToCreatePostPage}
              title="navigate to CreatePostPage"
              data-testid="picture-button"
            >
              <AiOutlinePicture styleName="post-creator-card__icon" />
            </button>
            <button
              styleName="post-creator-card__icon-link"
              onClick={navigateToCreatePostPage}
              data-testid="link-button"
            >
              <BsLink45Deg styleName="post-creator-card__icon" />
            </button>
          </div>
        </Card>
      )}
    </>
  );
};

export default CSSModules(PostCreatorCard, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "log",
});
