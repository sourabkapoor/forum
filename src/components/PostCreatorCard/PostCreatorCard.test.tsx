import { queryByPlaceholderText, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import PostCreatorCard from "./PostCreatorCard";
import { Provider } from "react-redux";
import StoreProvider from "../../redux/provider";
import { BrowserRouter } from "react-router-dom";

describe("PostCreatorCard", () => {
  test("snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <StoreProvider>
          <PostCreatorCard />
        </StoreProvider>
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  test("user is redirected to create post on input focus", async () => {
    render(
      <BrowserRouter>
        <StoreProvider>
          <PostCreatorCard />
        </StoreProvider>
      </BrowserRouter>
    );
    const mockNavigate = jest.fn();

    jest.mock("react-router-dom", () => ({
      ...jest.requireActual("react-router-dom"),
      useNavigate: () => mockNavigate,
    }));

    const input = screen.queryByPlaceholderText("Create Post");

    await userEvent.click(input as HTMLInputElement);

    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });
  test("user is directed to profile on profile picture click", async () => {
    render(
      <BrowserRouter>
        <StoreProvider>
          <PostCreatorCard />
        </StoreProvider>
      </BrowserRouter>
    );
    const mockNavigate = jest.fn();

    jest.mock("react-router-dom", () => ({
      ...jest.requireActual("react-router-dom"),
      useNavigate: () => mockNavigate,
    }));

    const profilePicture = screen.queryByTestId("profile-picture");

    await userEvent.click(profilePicture as HTMLDivElement);

    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });
  test("user is directed to AddPostForm page on image and link click", async () => {


    const mockNavigate = jest.fn();

    jest.mock("react-router-dom", () => ({
      ...jest.requireActual("react-router-dom") as any,
      useNavigate: () => mockNavigate,
    }));
    render(
      <BrowserRouter>
        <StoreProvider>
          <PostCreatorCard />
        </StoreProvider>
      </BrowserRouter>
    );
    const gallaryButton = screen.queryByTestId("picture-button");
    const linkButton = screen.queryByTestId("link-button");

    await userEvent.click(gallaryButton as HTMLButtonElement);
    await userEvent.click(linkButton as HTMLButtonElement);

    expect(mockNavigate).toHaveBeenCalledTimes(2);
  });
});
