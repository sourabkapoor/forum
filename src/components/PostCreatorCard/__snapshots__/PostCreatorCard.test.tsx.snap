// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCreatorCard snapshot 1`] = `<DocumentFragment />`;
